micro_dl/utils/tile_utils.py

```diff
@@ -10,34 +10,43 @@
 
 def read_imstack(input_fnames,
                  flat_field_fname=None,
-                 hist_clip_limits=None):
-    """Read the images in the fnames and assembles a stack
+                 hist_clip_limits=None,
+                 is_mask=False):
+    """
+    Read the images in the fnames and assembles a stack.
+    If images are masks, make sure they're boolean by setting >0 to True
     
     :param tuple input_fnames: tuple of input fnames with full path
     :param str flat_field_fname: fname of flat field image
     :param tuple hist_clip_limits: limits for histogram clipping
-    :return np.array: input stack flat_field correct and z-scored
+    :param bool: Indicator for if files contain masks
+    :return np.array: input stack flat_field correct and z-scored if regular
+        images, booleans if they're masks
     """
 
     im_stack = []
     for fname in input_fnames:
         im = read_image(fname)
         if flat_field_fname is not None:
             flat_field_image = np.load(flat_field_fname)
-            im = apply_flat_field_correction(
-                im,
-                flat_field_image=flat_field_image,
-            )
+            if not isinstance:
```

Comment:
should this be: if not ismask? or isinstance(?)

Reply:
I guess this is what happens when trying to code when you're sick. I'll try to apply some more logic on this in the afternoon. :-)

Reply:
Please rest and feel better! These changes can wait.